doc/kubeletConfig-check.md

```diff
@@ -0,0 +1,141 @@
+  namespace: openshift-compliance
+profiles:
+- apiGroup: compliance.openshift.io/v1alpha1
+  kind: Profile
+  name: ocp4-cis
+- apiGroup: compliance.openshift.io/v1alpha1
+  kind: Profile
+  name: ocp4-cis-node
+settingsRef:
+  apiGroup: compliance.openshift.io/v1alpha1
+  kind: ScanSetting
+  name: default
+```
+The `kubelteConfig` rules checks is executed against node file `/etc/kubernetes/kubelet.conf`.
+Depending on the roles a user has set in `ScanSetting`, checks will be executed against nodes with the selected node type.
+And in `ComplianceCheckResult`, `kubelteConfig` rules will be shown as `ocp4-cis-node-<role>-kubelet-*`.
+
+In the new approach, `KubeletConfig` rules belong to `ocp4-cis` profile. Compliance Operator checks runtime `KubeletConfig`
+through Kube `Node/Proxy` object, and it uses variables `ocp-var-role-master` and `ocp-var-role-master` to determine the nodes
+it performs the check against. And in `ComplianceCheckResult`, `kubelteConfig` rules will be shown as `ocp4-cis-kubelet-*`.
+The scan only passes if all selected nodes pass the check.
+
+## How to use it?
+
+In a standard cluster that only has `master` and `worker` pool, there isn't much needed to be done, everything will work out of the box.
+
+### a cluster that uses sub-pools
+
+In a cluster that has `master`, `worker` and `infra` pool, If a user wants to check against all pools, they need to set the value of
+`ocp-var-role-master` and `ocp-var-role-worker` to `infra` in `TailoredProfile`:
+
+```yaml
+apiVersion: compliance.openshift.io/v1alpha1
+kind: TailoredProfile
+metadata:
+  name: cis-infra-tp
+spec:
+  extends: ocp4-cis
+  title: My modified nist profile with a custom value
+  setValues:
+  - name: ocp4-var-role-master
+    value: infra
+    rationale: test for infra nodes
+  - name: ocp4-var-role-worker
+    value: infra
+    rationale: test for infra nodes
+  description: cis-infra-scan
+```
+And launch a scan using `ScanSettingBinding`:
+
+```yaml
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSettingBinding
+metadata:
+  name: cis
+  namespace: openshift-compliance
+profiles:
+- apiGroup: compliance.openshift.io/v1alpha1
+  kind: Profile
+  name: ocp4-cis
+- apiGroup: compliance.openshift.io/v1alpha1
+  kind: Profile
+  name: ocp4-cis-node
+- apiGroup: compliance.openshift.io/v1alpha1
+  kind: TailoredProfile
+  name: cis-infra-tp
+settingsRef:
+  apiGroup: compliance.openshift.io/v1alpha1
+  kind: ScanSetting
+  name: default
+```
+
+#### Remediation `KubeletConfig` on sub-pools
+
+If a user wants to use `KubeletConfig` remediation on sub-pools, they need to add a label to the sub-pool `MachineConfigPool`[1]:
+`$ oc label mcp infra pools.operator.machineconfiguration.openshift.io/infra=`
+
+[1] https://docs.openshift.com/container-platform/4.11/security/compliance_operator/compliance-operator-remediation.html#compliance-operator-apply-remediation-for-customized-mcp
+
+
+## List of variables used to be checked against `KubeletConfig`:
```

Comment:
This list could change, right? Do we have anything in place that's going to keep this updated in the event we add another variable? Otherwise, maybe we describe how users can derive this information instead of potentially documenting something that will become outdated.